### ISSUE TITLE
docker_container action redeploy should not start container which have never run

### ISSUE THREAD

**Issue by jperville** [bflad/chef-docker#432]
Docker containers are declared either with `action :create` or `action :run` (which is a shortcut to mean "create then start"). When a container has been created (never started) and then redeployed, I'd like `action :redeploy` to not start the container.

In other words, I'd like action redeploy to:
 - delete then run the container if it was previously running
 - delete then create the container if it was not previously running
So that the running state of the container after the redeploy is the same as it was before it.

In the current version (1.0.25), the docker_container `action :redeploy` is implemented like this:

```ruby
action :redeploy do
  action_delete
  action_run
end
```

In the older version (0.37) it has been implemented like this:

```ruby
action :redeploy do
  stop if (previously_running = running?)
  remove_container if exists?
  if previously_running
    run
  else
    create
  end
  new_resource.updated_by_last_action(true)
end
```

Would you take a PR to restore the behavior of version 0.37 (which was introduced in #242).

From reading the code, I know that I can tell if a container is running like this:
```ruby
c = Docker::Container.get(new_resource.container_name)
next if c.info['State']['Running']
```

**Comment by someara** [comment]
Indeed I would! Thanks!
-s

**Comment by jperville** [comment]
Fixed in PR #437.
